src/transformers/generation/candidate_generator.py

```diff
@@ -18,6 +18,7 @@
 
 import numpy as np
 import torch
+from sklearn.metrics import roc_curve
```

Comment:
sklearn is not installed by default, and we don't want to introduce a hard dependency to use assisted generation.

1 - we'll have to wrap this import in an import guard, i.e.
```py
from ..utils import is_sklearn_available
if is_sklearn_available():
    from sklearn.metrics import roc_curve
```
otherwise folks without the library will fail to initialize this module
2 - given that's an optional import and that this PR's technique "only" boosts speed, my suggestion would be to use the boost if the library is available, and don't use it if not. That way, assisted generation always works and the user is not forced to install an extra dependency 🤗 In the docs, we should recommend to install sklearn to benefit from this boost.

Reply:
@gante where do you suggest to  add the recommendation in the docs? 
[here](https://huggingface.co/docs/transformers/main/en/generation_strategies#speculative-decoding) ?

Reply:
@jmamou yes, that will be the correct place in docs